binary_search_tree/binary_search_tree.py

```diff
@@ -1,17 +1,60 @@
+from math import inf
+
+
+def equal_to(value):
+    def is_equal(something):
+        if something == value:
+            return True
+        else:
+            return False
+
+    enclosure = is_equal
+    return enclosure
+
+
 class BinarySearchTree:
-  def __init__(self, value):
-    self.value = value
-    self.left = None
-    self.right = None
+    def __init__(self, value):
+        self.value = value
+        self.left = None
+        self.right = None
+
+    def seek(self, value, cb=equal_to(None)):
+        branch = self
+        if value >= self.value:
+            if cb(self.right):
+                return branch
+            else:
+                branch = self.right
+                return branch.seek(value)
+        elif value < self.value:
+            if cb(self.left):
+                return self
+            else:
+                branch = self.left
+                return branch.seek(value)
+
+    def insert(self, value):
+        branch = self.seek(value)
+        # print(branch.value, branch.right, branch.left)
+        if value >= branch.value:
```

Comment:
If seek() did not find a node with the value, then branch is None and this method seems to fail and tests error.

we should be able to insert any value regardless of if it exists already in the tree or not.